Sources/joyfill-components-swift/Components/DateTime/RPicker.swift

```diff
@@ -0,0 +1,461 @@
+            if RPicker.sharedInstance.isPresented == false {
+                RPicker.sharedInstance.isPresented = true
+                
+                let vc = RPickerController(title: title, cancelText: cancelText, doneText: doneText, datePickerMode: datePickerMode, selectedDate: selectedDate, minDate: minDate, maxDate: maxDate, dataArray: dataArray, selectedIndex: selectedIndex, type: type, style: style)
+                
+                vc.modalPresentationStyle = .overCurrentContext
+                vc.modalTransitionStyle = .crossDissolve
+                cc.present(vc, animated: true, completion: nil)
+                
+                vc.onWillDismiss = {
+                    RPicker.sharedInstance.isPresented = false
+                }
+                return vc
+            }
+        }
+        return nil
+    }
+}
+
+class RPickerController: UIViewController {
+    
+    //MARK:- Public closuers
+    var onDateSelected : ((_ date: Date) -> Void)?
+    var onOptionSelected : ((_ value: String, _ atIndex: Int) -> Void)?
+    var onWillDismiss : (() -> Void)?
+    
+    //MARK:- Public variables
+    var selectedIndex: Int?
+    var selectedDate = Date()
+    var maxDate: Date?
+    var minDate: Date?
+    var titleText: String?
+    var cancelText: String?
+    var doneText: String = "Done"
+    var datePickerMode: UIDatePicker.Mode = .date
+    var datePickerStyle: RDatePickerStyle = .Wheel //Only for iOS 14 and above
+    var pickerType: RPickerType = .date
+    var dataArray: Array<String> = []
+    
+    //MARK:- Private variables
+    private let barViewHeight: CGFloat = 44
+    private let pickerHeight: CGFloat = 216
+    private let buttonWidth: CGFloat = 84
+    private let lineHeight: CGFloat = 0.5
+    private let buttonColor = UIColor(red: 72/255, green: 152/255, blue: 240/255, alpha: 1)
+    private let lineColor = UIColor(red: 240/255, green: 240/255, blue: 240/255, alpha: 1)
+    
+    //MARK:- Init
+    init(title: String? = nil,
+         cancelText: String? = nil,
+         doneText: String = "Done",
+         datePickerMode: UIDatePicker.Mode = .date,
+         selectedDate: Date = Date(),
+         minDate: Date? = nil,
+         maxDate: Date? = nil,
+         dataArray:Array<String> = [],
+         selectedIndex: Int? = nil,
+         type: RPickerType = .date,
+         style: RDatePickerStyle = .Wheel) {
+        
+        self.titleText = title
+        self.cancelText = cancelText
+        self.doneText = doneText
+        self.datePickerMode = datePickerMode
+        self.selectedDate = selectedDate
+        self.minDate = minDate
+        self.maxDate = maxDate
+        self.dataArray = dataArray
+        self.selectedIndex = selectedIndex
+        self.pickerType = type
+        self.datePickerStyle = style
+        
+        super.init(nibName: nil, bundle: nil)
+        initialSetup()
+    }
+    
+    required init?(coder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
```

Comment:
This can't work as global function because while creating a swift class of UIView type this function act as a initializer to that class so it is necessary to have this function in all the files.